### PR TITLE
Update README

### LINKED ISSUE #675: update README
i'd like to add some more dev-facing documentation to the README based on recent work we've done.

some ideas:

- [x] update section on traject config
- [x] add some examples of interacting with eye
- [x] explain our usage of kafka, add some examples of logging onto the kafka box and using the utilities there
- ~add a FOLIO section to "indexing strategies"~ (wait on this until things are more established)

more?

### PR DESCRIPTION
**Pull request by thatbudakguy**
Closes #675

Looking for review to make sure my descriptions of things are correct, and also add any other helpful commands/common dev tasks I may have missed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,40 +1,62 @@
...
+[![CI status](https://github.com/sul-dlss/searchworks_traject_indexer/actions/workflows/ruby.yml/badge.svg)](https://github.com/sul-dlss/searchworks_traject_indexer/actions/workflows/ruby.yml)
+[![Current release](https://img.shields.io/github/v/release/sul-dlss/searchworks_traject_indexer)](https://github.com/sul-dlss/searchwork_traject_indexer/releases)
+![tested on ruby 3.1](https://img.shields.io/badge/ruby-v3.1-red)
+
+<p align="center">indexing MARC, MODS, and more for <a href="https://github.com/sul-dlss/SearchWorks">SearchWorks</a>.</p>
+<img src="preview.png" align="center" alt="solr index fields displayed overlaid on SearchWorks catalog preview for a book">
 
-## Installation
+## local development
+searchworks_traject_indexer is built on the [traject](https://github.com/traject/traject) transformation library, which requires ruby. we test the application using ruby 3.1; support for other versions is not guaranteed.
```
Comment (cbeer) [inline README.md:10]: and jruby

File: README.md [hunk 2]
```diff
@@ -48,30 +70,51 @@ processes:
...
+other configuration values, like the URL of the solr instance, are usually set at the top of the configuration file using traject's `provide`. many can be set by environment variables; some of these in turn are set by the eye configuration:
+```yaml
+processes:
+  - name: my_indexer
+    env:
+      SOLR_URL: http://sul-solr.stanford.edu/solr/searchworks-prod # controls where traject indexes to
+```
+in local development, you can invoke traject on the command line and override these settings by passing environment variables:
+```sh
+SOLR_URL=http://localhost:8983/solr/core-name bundle exec traject -c lib/traject/config/sirsi_config.rb my_marc_file.marc
```
Comment (marlo-longley) [inline README.md:108]: I found it helpful to know that you can do things like:
`SOLR_URL=http://localhost:8983/solr/core-name bundle exec traject -c lib/traject/config/sirsi_config.rb -w Traject::DebugWriter /some/path/file.ext` to output data for debugging

Also, and this is probably too specific for a general README, but I also debugged using the JSON writer
`SOLR_URL=http://localhost:8983/solr/core-name bundle exec traject -c lib/traject/config/sirsi_config.rb -w Traject::JsonWriter /Users/marloelilongley/Downloads/2306638.marc | tail -n +2 | jq '.toc_struct'` (removes the honeybadger line and then puts the section of output you want into something readable)

**Review by marlo-longley — APPROVED** [review]
Thanks for taking the time to document all these processes @thatbudakguy ! It was definitely helpful for me to just read over this.

File: README.md (hunk 2 above)
Comment (thatbudakguy) [reply README.md:108]: whoa, i didn't even know you could do this! this definitely belongs in the readme.

File: README.md (hunk 1 above)
Comment (thatbudakguy) [reply README.md:10]: good catch, thanks

**Pull request merged**